compiler/qsc_codegen/src/qir.rs

```diff
@@ -314,6 +352,31 @@ fn call_to_qir(
     }
 }
 
+fn fcmp_to_qir(
+    op: FcmpConditionCode,
+    lhs: &rir::Operand,
+    rhs: &rir::Operand,
+    variable: rir::Variable,
+    program: &rir::Program,
+) -> String {
+    let lhs_ty = get_value_ty(lhs);
+    let rhs_ty = get_value_ty(rhs);
+    let var_ty = get_variable_ty(variable);
+    assert_eq!(
+        lhs_ty, rhs_ty,
+        "mismatched input types ({lhs_ty}, {rhs_ty}) for icmp {op}"
```

Comment:
Maybe I'm misreading this, but why does the message here (and line 370) refer to `icmp` if this is inside the `fcmp` method?